extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,886 @@
+    if (!body) {
+      throw new Error("No response body found");
+    }
+
+    const archiveFileStream = createWriteStream(archivePath);
+
+    const contentLength = response.headers.get("content-length");
+    const totalNumBytes = contentLength
+      ? parseInt(contentLength, 10)
+      : undefined;
+
+    const reportProgress = reportStreamProgress(
+      "Downloading source root...",
+      totalNumBytes,
+      progressCallback,
+    );
+
+    try {
+      const reader = body.getReader();
+      for (;;) {
+        const { done, value } = await reader.read();
+        if (done) {
+          break;
+        }
+
+        onData();
+        reportProgress(value?.length ?? 0);
+
+        await new Promise((resolve, reject) => {
+          archiveFileStream.write(value, (err) => {
+            if (err) {
+              reject(err);
+            }
+            resolve(undefined);
+          });
+        });
+      }
+
+      await new Promise((resolve, reject) => {
+        archiveFileStream.close((err) => {
+          if (err) {
+            reject(err);
+          }
+          resolve(undefined);
+        });
+      });
+    } catch (e) {
+      // Close and remove the file if an error occurs
+      archiveFileStream.close(() => {
+        void remove(archivePath);
+      });
+
+      if (e instanceof DOMException && e.name === "AbortError") {
+        const thrownError = new Error("The download timed out.");
+        thrownError.stack = e.stack;
+        throw thrownError;
+      }
+
+      throw new Error(
+        `Error downloading source root. Reason: ${getErrorMessage(e)}`,
+      );
+    } finally {
+      disposeTimeout();
+    }
+
+    void extLogger.log(`Download complete, extracting source...`);
+
+    // Extract the downloaded zip file
+    await unzipToDirectoryConcurrently(
+      archivePath,
+      outputPath,
+      progressCallback
+        ? reportUnzipProgress(`Unzipping source root...`, progressCallback)
+        : undefined,
+    );
+    await remove(archivePath);
+
+    return checkoutDir;
+  } catch (error) {
+    throw new Error(
+      `Failed to download ${nwo} at ${sha}:. Reason: ${getErrorMessage(error)}`,
+    );
+  }
+}
+
+/**
+ * Runs autofix for a given repository (nwo).
+ */
+async function runAutofixForRepository(
+  nwo: string,
+  sarifFile: string,
+  srcRootPath: string,
+  localAutofixPath: string,
+  autofixOutputStoragePath: string,
+  resultCount: number,
+  outputTextFiles: string[],
+): Promise<void> {
+  // Get storage paths for the autofix results for this repository.
+  const {
+    repoAutofixOutputStoragePath,
+    outputTextFilePath,
+    transcriptFilePath,
+    fixDescriptionFilePath,
+  } = await getRepoStoragePaths(autofixOutputStoragePath, nwo);
+
+  // Get autofix binary.
+  // Switch to Go binary in the future and have user pass full path
+  // in an environment variable instead of hardcoding part here.
+  const cocofixBin = join(process.cwd(), localAutofixPath, "bin", "cocofix.js");
```

Comment:
If the Go binary accepts the same parameters as cocofix, would it makes sense to already add a config setting for this and use a default of `bin/cocofix.js`?